main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
```

Comment:
`filepath.Walk()` is probably more expensive than `ioutil.ReadDir()` or even `filepath.Glob()`. It also seems overkill to recursively traverse the directory.

Reply:
Not sure how ioutil.ReadDir() can be less expensive? filepath.Walk() offers convenient api to process the file at the same time. This is a directory with just a few files and this only runs once.